src/app/controllers/albumDetailController.js

```diff
@@ -0,0 +1,39 @@
+export function albumDetailController($scope, $routeParams, $location, ApiService) {
+    $scope.albumDetailCtrl = this;
+
+    ApiService.getTracks($routeParams.albumId)
+    .then(function(response) {
+        this.tracks = response;
+    }.bind(this));
+
+    ApiService.getArtist($routeParams.bandId)
+    .then(function(response) {
+        this.artist = response;
+    }.bind(this));
+
+    ApiService.getAlbum($routeParams.albumId)
+    .then(function(response) {
+        this.album = response;
+    }.bind(this));
+
+    this.toIndex = function() {
+        $location.path("/");
+    }.bind(this);
+
+    this.toResults = function() {
+        $location.path("/results/" + this.search);
+    }.bind(this);
+
+    this.toBandAlbums = function() {
+        $location.path("/band-albums/" + this.artist.id);
+    }.bind(this);
+
+    this.backToResults = function() {
+        $location.path("/results/" + this.artist.name);
+    }.bind(this);
+
+    this.playTrack = function(track) {
```

Comment:
Ok, but if you change of view, the song will still be played, check this article and the controller hooks in order to stop the song playing when controller has been destroyed. http://blog.thoughtram.io/angularjs/2016/03/29/exploring-angular-1.5-lifecycle-hooks.html